hcl2template/testdata/datasources/recursive.pkr.hcl

```
@@ -0,0 +1,15 @@
+data "null" "foo" {
+  input = "chocolate"
+}
+
+data "null" "bar" {
+  input = "vanilla"
+}
+
+data "null" "baz" {
+  input = "${data.null.foo.output}-${data.null.bar.output}-swirl"
+}
+
```

What about adding one that it's in the wrong order? 
```suggestion
data "null" "yummy" {
  input = "${data.null.bang.output}-and-cinamon"
}
```

---

Nice idea!

---

I should also do one that actually triggers the max recursion depth, and one that reads data sources from other files.

---

Oh yeah, that will be nice!

---

I believe this is already part of the datasouce agreement, but I would add a test for a set of data sources that depend on each other. This should be an error right? 

```
data "null" "gummy" {
  input = "${data.null.bear.output}"
}
data "null" "bear" {
  input = "${data.null.gummy.output}"
}
```

---

Great call -- I think that'll trigger the "max recursion depth" I referenced above, so it'll knock out both cases.

---

Done!